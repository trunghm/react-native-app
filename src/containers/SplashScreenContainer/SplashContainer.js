'use strict'

import React from 'react'
import {connect} from 'react-redux'
import SplashComponent from './views/index'

const mapStateToProps = (state) => ({
     userState: state.testReducer,
})

// Define the actions this component may dispatch
const mapDispatchToProps = (dispatch) => {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SplashComponent)
