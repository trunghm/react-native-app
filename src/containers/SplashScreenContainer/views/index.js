import SplashComponent from './SplashComponent'
export  default  SplashComponent
