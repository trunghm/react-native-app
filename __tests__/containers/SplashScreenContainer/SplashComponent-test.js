/**
 * @format
 */

import 'react-native';
import React from 'react';
import {SplashComponent} from '../../../src/containers/SplashScreenContainer/SplashContainer';

// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  SplashComponent component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <SplashComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <SplashComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

});
