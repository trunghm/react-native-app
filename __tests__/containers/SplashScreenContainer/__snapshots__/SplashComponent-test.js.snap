// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  SplashComponent component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  SplashComponent component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/themes/images/splash.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
